tailcfg/tailcfg.go

```diff
@@ -1223,6 +1223,26 @@ const (
 // the application via the WhoIs API.
 type PeerCapMap map[PeerCapability][]json.RawMessage
 
+// UnmarshalCapJSON unmarshals the cap values into the provided type.
+// It returns nil, nil if the capability is not present.
+// It returns an error if the values cannot be unmarshaled into the provided
+// type.
+func UnmarshalCapJSON[T any](cm PeerCapMap, cap PeerCapability) ([]*T, error) {
```

Comment:
this is to assume structs (JSON objects) too much. didn't we design the ACLs such that the args can be JSON objects, strings, numbers, or anything except lists? for strings, a slice of pointers to strings would be weird.

Return `[]T` instead and let caller choose to pass `*T` if they want that?

Reply:
We restricted a single grant to not be a JSON array, but each rule can have multiple grants and there are multiple rules. So at this point we were going to get a JSON array. Our reasoning to disallow JSON arrays in the ACL was specifically to avoid the case of `[[..], [.], [...]]` as thats a bit confusing.

return `[]T` sgtm, done.